### ISSUE TITLE
ValueError: Inferred frequency MS from passed dates does not conform to passed frequency M

### ISSUE THREAD

**Issue by yarikoptic** [statsmodels/statsmodels#3283]
With fresh pandas/numpy/... in debian sid we see
```
ERROR: statsmodels.graphics.tests.test_tsaplots.test_plot_month
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/build/statsmodels-0.8.0~rc1+git59-gef47cd9/debian/python-statsmodels/usr/lib/python2.7/dist-packages/statsmodels/graphics/tests/test_tsaplots.py", line 107, in test_plot_month
    dta.index = pd.DatetimeIndex(dates, freq='M')
  File "/usr/lib/python2.7/dist-packages/pandas/util/decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/pandas/tseries/index.py", line 385, in __new__
    .format(inferred, freq.freqstr))
ValueError: Inferred frequency MS from passed dates does not conform to passed frequency M
```
some detail:
```
380                                                 freq, tz=tz, ambiguous=ambiguous)
381                         if not np.array_equal(subarr.asi8, on_freq.asi8):
382                             raise ValueError('Inferred frequency {0} from passed '
383                                              'dates does not conform to passed '
384                                              'frequency {1}'
385  ->                                          .format(inferred, freq.freqstr))
386     
387             if freq_infer:
388                 inferred = subarr.inferred_freq
389                 if inferred:
390                     subarr.offset = to_offset(inferred)
(Pdb) p subarr.asi8
array([-631152000000000000, -628473600000000000, -626054400000000000,
       -623376000000000000, -620784000000000000, -618105600000000000,
       -615513600000000000, -612835200000000000, -610156800000000000,
       -607564800000000000, -604886400000000000, -602294400000000000,
```

**Comment by josef-pkt** [comment]
What's the status on this one?
I guess that's part of the refactoring in master for 0.9, but I have not idea if there is a resolution with 0.8.

I'm checking some debian testing, as far as I find those issues.

**Comment by ChadFulton** [comment]
This is unrelated to the refactoring I did in 0.9, and in fact is not fixed by it. I did not touch any of the `statsmodels.graphics.tsaplots` in the refactoring (my changes only affected `TimeSeriesModel` and descendents).

This is just an error in the unit test itself. Pandas' frequency `"M"` corresponds to month end and the unit test constructs dates like "1 2016 JAN" which should be month start. I don't know why the unit test ever worked - maybe when it was added (in 2014) Pandas was more forgiving about month-related coercion?

**Comment by ChadFulton** [comment]
I'll make a PR and backport PR to fix it.

**Comment by ChadFulton** [comment]
I have PRs for this which seem to work, although I don't see this test failure on the recent travis builds. I was able to replicate the test failure on my machine with recent Pandas, though, and the PRs fix it for me.

**Comment by josef-pkt** [comment]
Given that you can replicate it, a fix would be very good.

One problem is that there are too many versions of pandas and numpy with possibly small differences and we don't test against all of those versions on TravisCI. 
(Debian testing still shows the file closed error, when they use pandas 0.19.0, but not with 0.19.1)

(aside: I have no idea if we need to adjust anything for python 3.6)

**Comment by ChadFulton** [comment]
Guess I'll merge then!
